### ISSUE TITLE
leveldb for iOS does not compile with Xcode 4.3

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmithdemos/leveldb#86]
```
What steps will reproduce the problem?
1. Install Xcode 4.3
2. make PLATFORM=IOS

What is the expected output? What do you see instead?

Expected output: 

leveldb compiles

Actual output:

2012-05-02 21:59:53.852 defaults[69258:303] 
The domain/default pair of (/Developer/Platforms/iPhoneOS.platform/version, 
CFBundleShortVersionString) does not exist
2012-05-02 21:59:53.872 defaults[69260:303] 
The domain/default pair of (/Developer/Platforms/iPhoneOS.platform/version, 
CFBundleShortVersionString) does not exist
mkdir -p ios-x86/db/
2012-05-02 21:59:53.889 defaults[69272:303] 
The domain/default pair of (/Developer/Platforms/iPhoneOS.platform/version, 
CFBundleShortVersionString) does not exist
2012-05-02 21:59:53.905 defaults[69273:303] 
The domain/default pair of (/Developer/Platforms/iPhoneOS.platform/version, 
CFBundleShortVersionString) does not exist
/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin/c++ -I. 
-I./include -fno-builtin-memcmp -DOS_MACOSX -DLEVELDB_PLATFORM_POSIX -O2 
-DNDEBUG        -isysroot 
/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk
 -arch i686 -c db/builder.cc -o ios-x86/db/builder.o
make: /Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin/c++: No 
such file or directory
make: *** [db/builder.o] Error 1
make: *** Waiting for unfinished jobs....

What version of the product are you using? On what operating system?

leveldb git @ 85584d497, Mac OS X 10.7.3, Xcode 4.3.2

Please provide any additional information below.

The attached patch resolves the problem.

(Note: I'm a Facebook employee. I previously contributed to Google 
libphonenumber, and I filed a Corporate CLA with Google for libphonenumber 
which should apply here.)
```

---

Original issue reported on code.google.com by `bgertzfi...@gmail.com` on 3 May 2012 at 5:04

Attachments:
- [leveldb-xcode-4.3.patch](https://storage.googleapis.com/google-code-attachments/leveldb/issue-86/comment-0/leveldb-xcode-4.3.patch)

**Comment by GoogleCodeExporter** [comment]
```
Can you try against leveldb-1.5.0?  That contains a bunch
of build related fixes, though not your patch.

```

---

Original comment by `san...@google.com` on 30 May 2012 at 5:10

**Comment by GoogleCodeExporter** [comment]
---

Original comment by `san...@google.com` on 30 May 2012 at 5:34
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Confirmed fixed in leveldb-1.5.0. Thanks!
```

---

Original comment by `bgertzfi...@gmail.com` on 30 May 2012 at 10:56
